ssl/quic/quic_impl.c

```diff
@@ -3446,14 +3512,13 @@ QUIC_NEEDS_LOCK
 static int qctx_should_autotick(QCTX *ctx)
 {
     int event_handling_mode;
+    QUIC_OBJ *obj = ctx->obj;
 
-    if (ctx->is_stream) {
-        event_handling_mode = ctx->xso->event_handling_mode;
-        if (event_handling_mode != SSL_VALUE_EVENT_HANDLING_MODE_INHERIT)
-            return event_handling_mode != SSL_VALUE_EVENT_HANDLING_MODE_EXPLICIT;
-    }
+    for (; (event_handling_mode = obj->event_handling_mode)
+            == SSL_VALUE_EVENT_HANDLING_MODE_INHERIT
+            && obj->parent_obj != NULL;
+         obj = obj->parent_obj);
```

Comment:
I'd prefer keeping the first condition on a single line regardless of the 80 char limit. The current split makes it really badly readable.

```suggestion
    for (; (event_handling_mode = obj->event_handling_mode) == SSL_VALUE_EVENT_HANDLING_MODE_INHERIT
            && obj->parent_obj != NULL;
         obj = obj->parent_obj);
```